blog/_src/posts/2017-09-27-tutorial-contributing-to-racket.md

```diff
@@ -0,0 +1,414 @@
+```
+<br/>
+
+In DrRacket, you can change the "Submodules to Run" setting.
+
+
+### Running package-wide tests
+
+For many packages, the following command runs all tests:
+
+```
+$ raco test -c <PKG>
+```
+<br/>
+
+Other repos have special commands for running tests.
+The [`racket`](https://github.com/racket/racket)
+ and [`typed-racket`](https://github.com/racket/typed-racket)
+ are two examples.
+For these, look at the scripts `.travis.yml` and/or `appveyor.yml` to find
+ the exact commands to run.
+
+These CI tests will run when you push the pull request, but it's good to check
+ before you push.
+
+
+### 5) Create a PR
+
+To create the pull request, first push the changes from your clone to your fork:
+
+```
+$ git push origin my-branch
+```
+<br/>
+
+Second, use GitHub to create a pull request.
+If you visit the package's repo just after pushing your branch to your fork,
+ there should be a green button that you can click to make the pull request.
+If not, follow GitHub's [guide on creating a pull request](<https://help.github.com/articles/creating-a-pull-request/#creating-the-pull-request>).
+
+Just before the pull request is public, you'll have a chance to pick a title,
+ add a summary, and review the code.
+
+
+- - -
+
+# FAQ
+
+## Q. I just submitted a pull request. What happens now?
+
+Now you enter the "LGTM loop".
+
+```
+do {
+  sleep($a_few_hours);
+  if ($changes_requested) {
+    make_changes();
+  }
+  if ($one_week_with_no_comments) {
+    send_gentle_reminder();
+  }
+} until ($LGTM);
+```
+
+(This code is in Perl so you know not to take it too seriously.)
+
+- Wait a few days for someone to review the pull request.
+  You should receive an email when this happens.
+- If a week goes by with no comment, consider sending a quick "ping, any
+  thoughts on this?" comment to the pull request.
+- If someones comments and requests changes, then make changes in a new commit
+  and push to your branch. GitHub will detect these changes and update the
+  pull request.
+
+Eventually, someone will comment "LGTM" and merge your changes.
+
+"LGTM" probably means "looks good to merge", but the exact meaning depends
+ on who wrote it. No matter what, it's a compliment.
+
+
+## Q. Is there an easy way to submit typo fixes?
+
+If you want to submit a small typo fix, then you can follow GitHub's guide
+ on [_Editing files in another user's repository_](https://help.github.com/articles/editing-files-in-another-user-s-repository/).
+
+The guide assumes you know the **repository** and **file** you want to change.
+Finding the repository should be easy; follow the hints above.
+Finding the file may be difficult, because package documentation is generated
+ from Scribble files.
+
+If the hints above fail you, a "last resort" option is to clone the package
+ and use your operating system to search for the right file.
+That, or ask someone for help.
+
+
+## Q. Do I need to install Racket from source to contribute?
```

Comment:
Based on length & complexity, this is seeming like it deserves to be a section in the docs, not just a blog post?